lighthouse-core/lib/url-shim.js

```diff
@@ -215,6 +215,32 @@ class URLShim extends URL {
     const urlScheme = protocol.includes(':') ? protocol.slice(0, protocol.indexOf(':')) : protocol;
     return NON_NETWORK_SCHEMES.includes(urlScheme);
   }
+
+  /**
+   * @param {string} src
+   * @return {string|undefined}
+   */
+  static guessMimeType(src) {
+    let url;
+    try {
+      url = new URL(src);
+    } catch {
+      return undefined;
+    }
+
+    if (url.protocol === 'data:') {
+      const match = url.pathname.match(/image\/(png|jpeg|jpg|svg\+xml|webp|gif|avif);/);
+      if (!match) return undefined;
+      return match[0].split(';')[0];
+    }
+
+    const match = url.pathname.toLowerCase().match(/\.(png|jpeg|jpg|svg|webp|gif|avif)$/);
+    if (!match) return undefined;
+
+    const ext = match[1];
+    if (ext === 'svg') return 'image/svg+xml';
```

Comment:
```suggestion
    let ext = match[1];
    if (ext === 'jpg') ext = 'jpeg';
    if (ext === 'svg') return 'image/svg+xml';
```

or explicit return? doesn't really matter to me